aten/src/ATen/native/cpu/MaxPooling.cpp

```diff
@@ -0,0 +1,56 @@
+#include <ATen/ATen.h>
+#include <ATen/Parallel.h>
+#include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/MaxPooling.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+template <typename scalar_t>
+inline void max_pool1d_kernel(
+    scalar_t* op,
+    const scalar_t* ip,
+    const PoolingParams1D& p) {
+  for (int64_t kj = 0; kj < p.KW; ++kj) {
+    int64_t oj = p.valid_kernel_start(kj);
+    int64_t oe = p.valid_kernel_end(kj);
+    int64_t ij = oj * p.SJ + kj * p.DJ - p.PJ;
+    for (; oj < oe; ++oj, ij += p.SJ) {
+      bool update_max = std::isnan(ip[ij]) || op[oj] < ip[ij];
+      op[oj] = update_max ? ip[ij] : op[oj];
+    }
+  }
+}
+
+void max_pool1d_impl(
+    Tensor& output,
+    const Tensor& input,
+    const PoolingParams1D& p) {
+  AT_DISPATCH_FLOATING_TYPES(input.scalar_type(), "max_pool1d_impl", [&] {
```

Comment:
nit: This is the same as current pooling function. I am wondering could we extend this to integer as well? Any blocking issue for max pooling to support integer?

Reply:
There shouldn't be any problem with extending to integers, I'll change it.